### PR TITLE
add configuration for SequoiaDB authentication

### PR DESCRIPTION
**Pull request by mood1994**
I add two configurations, sdb_user and sdb_password, which can make mysql to connect sequoiadb with authentication, and the password is hidden when show variables.

### CODE REVIEW

File: ha_sdb.cc [hunk 1]
```diff
@@ -2084,18 +2095,15 @@ static int sdb_done_func(void *p) {
...
-    PLUGIN_LICENSE_GPL,
-    sdb_init_func, /* Plugin Init */
-    sdb_done_func, /* Plugin Deinit */
-    0x0300,        /* version */
-    NULL,          /* status variables */
-    sdb_sys_vars,  /* system variables */
-    NULL,          /* config options */
-    0,             /* flags */
-} mysql_declare_plugin_end;
+mysql_declare_plugin(sequoiadb) {
```
Comment (davidli2010) [inline ha_sdb.cc]: Do not change the code format.

File: sdb_util.cc [hunk 2]
```diff
@@ -15,10 +15,12 @@
 
 #include "sdb_util.h"
 #include <sql_table.h>
-#include <string.h>
 #include "sdb_log.h"
 #include "sdb_errcode.h"
 #include "sdb_def.h"
+#include <my_rnd.h>
+
+Sdb_encryption sdb_encryption;
```
Comment (davidli2010) [inline sdb_util.cc]: It's bad to define the instance of Sdb_encryption  here. 
sdb_util.cc/.h is only for utilities code, do not define variable in it.
It's better define this variable in sdb_conf and rename to sdb_password_encryption as it's used by password.

File: ha_sdb.cc [hunk 3]
```diff
@@ -2070,6 +2071,16 @@ static int sdb_init_func(void *p) {
     return 1;
   }
 
+  String src_password(sdb_password, &my_charset_bin);
```
Comment (davidli2010) [inline ha_sdb.cc]: Encapsulate the encrytion process in a function in sdb_conf, and need to extern src_password.

Comment (davidli2010) [reply ha_sdb.cc]: I make a mistake. Here I mean it's **no need** to extern src_password.

**Comment by mood1994** [comment]
Ok, the problems above have been solved. Check it out.

**Pull request merged**